client/src/app/pages/assessment-management/questionnaire/components/answer-table.tsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { Table, Tbody, Td, Th, Thead, Tr } from "@patternfly/react-table";
+
+import { useLocalTableControls } from "@app/hooks/table-controls";
+import {
+  ConditionalTableBody,
+  TableHeaderContentWithControls,
+  TableRowContentWithControls,
+} from "@app/components/TableControls";
+import { NoDataEmptyState } from "@app/components/NoDataEmptyState";
+import { Answer } from "@app/api/models";
+import { Label, Text } from "@patternfly/react-core";
+import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+import { IconedStatus, IconedStatusPreset } from "@app/components/IconedStatus";
+
+export interface IWaveStatusTableProps {
+  answers: Answer[];
+}
+
+const AnswerTable: React.FC<IWaveStatusTableProps> = ({ answers }) => {
+  const { t } = useTranslation();
+
+  const tableControls = useLocalTableControls({
+    idProperty: "choice",
+    items: answers,
+    columnNames: {
+      choice: "Answer choice",
+      weight: "Weight",
+    },
+    hasActionsColumn: false,
+    hasPagination: false,
+    variant: "compact",
+  });
+  const {
+    currentPageItems,
+    numRenderedColumns,
+    propHelpers: { tableProps, getThProps, getTdProps },
+  } = tableControls;
+  const mapRiskToStatus = (risk: string): IconedStatusPreset => {
+    switch (risk) {
+      case "green":
+        return "LowRisk";
```

Comment:
if you wanted to not add to the `IconedStatus` component, you could just return from here...

"green" = `<IconedStatus preset="ok" />` (as it already matches that preset)
"unknown" = `<IconedStatus preset="unknown" />`
"red" = `<IconedStatus icon={<TimesCircleIcon />} status="danger" />`
"yellow" = `<IconedStatus icon={<WarningTriangleIcon />} status="warning" />`

and then line 88 could be just `{ mapRiskToStatus(answer.risk) }` .. although maybe rename to `getIconByRisk`

but more of a matter of preference and if we expect these presets to be reused

Reply:
Done!